compose/config/config.py

```diff
@@ -217,20 +282,20 @@ def merge_services(base, override):
             for name in all_service_names
         }
 
-    config_file = process_config_file(config_details.config_files[0])
-    for next_file in config_details.config_files[1:]:
-        next_file = process_config_file(next_file)
-
+    config_file = config_files[0]
+    for next_file in config_files[1:]:
         config = merge_services(config_file.config, next_file.config)
         config_file = config_file._replace(config=config)
 
     return build_services(config_file)
 
 
-def process_config_file(config_file, service_name=None):
+def process_config_file(config_file, service_name=None, version=None):
```

Comment:
I think `ServiceExtendsResolver.validate_and_construct_extends()` will need to call this with a version as well. 

Since `version` is a required parameter can we make it a position arg instead of a kwarg?